### DISCUSSION TITLE
Example for using (local) indexes

### DISCUSSION THREAD

**Discussion by johannisjo** [sensedeep/dynamodb-onetable#39] [Q&A] [answered]
Hi,

I'm testing this library at the moment, to see if I can use it in an application at work. I have some trouble following how to use and query indexes. In this specific case I'm interested in using a local secondary index (LSI).

The documentation mentions that my indexes will be defined as LSIs "if they only contain as hash key", and then goes on to mention "Omit the hash for an LSI" in a code comment. I suspect the latter is correct? However, when I query my table, I keep getting an error: `dynamo: Empty hash key`. I suspect I am using it wrong. Do you have any examples of using LSIs?

**Comment by mobsense** [comment, marked as answer]
We've updated the doc in GitHub README.

In the schema, if you set the LSI index to have the same hash as the primary hash value, or if you omit the hash value. We assume it is an LSI. If the hash value is different to the primary, it is a GSI.

Now, when you use an API, you must of course provide the hash and sort key for the LSI. Both are required by DynamoDB for an LSI.
